Add Thai word list from Thai Wikipedia titles

konbraphat51 opened the pull request:
### What does this changes

Add an optional corpus of Wikipedia titles.
* thai_wikipedia_titles() function
* license description

Fixes #858 

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](https://github.com/PyThaiNLP/pythainlp/blob/dev/CONTRIBUTING.md) to this repository.

- [x] Passed code styles and structures
- [x] Passed code linting checks and unit test

konbraphat51 committed:
Add: wikipedia titles data

from https://github.com/konbraphat51/Thai_Dictionary_Cleaner

konbraphat51 committed:
Change file name

for clarification

konbraphat51 committed:
Add: wikipedia_titels()

function to return frozen set of wikipedia titles.
black used.

konbraphat51 committed:
Add: init redirection for wikipedia_titles()

konbraphat51 committed:
Add: testcode of wikipedia_titles()

konbraphat51 committed:
Add: license description

konbraphat51 commented:
Review please

konbraphat51 committed:
Add: empty line

obeying to code rules

bact committed:
Small editorial changes to wikipedia_titles.py

bact commented:
Looks good to me.

bact commented on tests/test_corpus.py:
```diff
@@ -25,6 +25,7 @@
     thai_syllables,
     thai_synonym,
     thai_words,
+    wikipedia_titles,
```
Do you mind to put it in alphabetical order please?

konbraphat51 replied:
Oh sorry, fixed!

bact reviewed (changes requested):
Code, docs, license, test. All looks good.

Once sorted out the order of module imports, I can approve this.

bact committed:
Update docstring in wikipedia_titles.py

bact commented on pythainlp/corpus/corpus_license.md:
```diff
@@ -101,3 +101,34 @@ S. Thoongsup et al., ‘Thai WordNet construction’,
+## Wikipedia Titles
+Corpus of Wikipedia titles (wikipedia_titles.txt) was processed by konbraphat51 (https://github.com/konbraphat51/Thai_Dictionary_Cleaner/tree/main)
+
+The original data is thwiki-latest-all-titles.gz of https://dumps.wikimedia.org/thwiki/latest/ which Wikipedia.org has created.
```
Do you mind to add date of the Wikipedia data here please?
The date that you have downloaded the data for your preparation.

konbraphat51 replied:
Yeah that's definitely required. Added!

bact committed:
Update wikipedia_titles.py

konbraphat51 committed:
Reorder import

konbraphat51 committed:
Added date of the data

konbraphat51 commented:
Okey, review again please

wannaphong reviewed (approved):
It look great!

bact reviewed (approved):
Thank you.

konbraphat51 committed:
Merge pull request #7 from PyThaiNLP/dev

pull

konbraphat51 committed:
Merge branch 'dev' into corpus_wiki

konbraphat51 committed:
Merge branch 'corpus_wiki' of https://github.com/konbraphat51/pythainlp into corpus_wiki

konbraphat51 committed:
Fix: duplication

konbraphat51 commented:
Ok, handled with the conflict!

bact committed:
Update corpus_license.md

For standard licenses, like Creative Commons, just link to the license URL. No need to put the license text inside this file.

bact committed:
Update test_corpus.py

- sort imports
- clean up test structure

bact committed:
Sort imports __init__.py

bact committed:
Update wikipedia_titles.py

bact commented:
Merged. Thank you.
